src/main/java/org/broadinstitute/hellbender/tools/diagnostics/PrintBGZFBlockInformation.java

```diff
@@ -0,0 +1,189 @@
+package org.broadinstitute.hellbender.tools.diagnostics;
+
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.tribble.AbstractFeatureReader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * A diagnostic tool that prints information about the compressed blocks in a BGZF format file,
+ * such as a .vcf.gz file.
+ *
+ * The output looks like this:
+ *
+ * Block at file offset 0
+ *     - compressed size: 12932
+ *     - uncompressed size: 65280
+ *
+ * Block at file offset 12932
+ *     - compressed size: 9978
+ *     - uncompressed size: 65280
+ * ...
+ * etc.
+ *
+ * The output can be redirected to a file using the -O option.
+ */
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = "Print information about the compressed blocks in a BGZF format file",
+        oneLineSummary = "Print information about the compressed blocks in a BGZF format file",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class PrintBGZFBlockInformation extends CommandLineProgram {
+    private final Logger logger = LogManager.getLogger(PrintBGZFBlockInformation.class);
+
+    @Argument(fullName = "bgzf-file", doc = "The BGZF-format file for which to print block information", optional = false)
+    private String bgzfPathString;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "File to which to write block information (if not specified, prints to standard output", optional = true)
+    private String output;
+
+    private Path bgzfPath;
+
+    private long streamOffset = 0l;
+
+    private PrintStream outStream;
+
+    @Override
+    protected void onStartup() {
+        super.onStartup();
+
+        bgzfPath = IOUtils.getPath(bgzfPathString);
+
+        if ( ! Files.exists(bgzfPath) ) {
+            throw new UserException("File " + bgzfPathString + " does not exist");
```

Comment:
Maybe better a `UserException.CouldNotReadInputFile` to narrow down?

Reply:
Changed to `UserException.CouldNotReadInputFile`